src/detect-transform-base64.c

```diff
@@ -37,33 +37,48 @@
 #include "util-unittest.h"
 #include "util-print.h"
 
-static int DetectTransformFromBase64DecodeSetup(DetectEngineCtx *, Signature *, const char *);
-static void DetectTransformFromBase64DecodeFree(DetectEngineCtx *, void *);
 #ifdef UNITTESTS
 #define DETECT_TRANSFORM_FROM_BASE64_MODE_DEFAULT (uint8_t) SCBase64ModeRFC4648
 static void DetectTransformFromBase64DecodeRegisterTests(void);
 #endif
-static void TransformFromBase64Decode(
-        DetectEngineThreadCtx *det_ctx, InspectionBuffer *buffer, void *options);
+#define B64D_ID_DATA_LEN 128
 
-void DetectTransformFromBase64DecodeRegister(void)
+static const char *id_format = "{"
```

Comment:
I would rather go for some binary encoding than a string...
Why did you pick this ?

Reply:
The actual coding is opaque so yes, it could be binary or something not directly printable. The reason it's a string is that it is hoped to be included in `--engine-analysis` output.

Reply:
`--engine-analysis` will use a jsonbuilder and not this format string, right ?

Reply:
Originally, the `id` member was meant to further describe the transform in a transform-specific way. Currently, it jsonifies the base64 transform and the others use the transform options. However, the value used for id can be anything.

Once the structural pieces are in place, we are free to explore different representations, including using the json-builder logic to construct the identity string and feed `--engine-analysis`

Reply:
> Once the structural pieces are in place, we are free to explore different representations

Not sure I agree on this